src/Management/CodeGenerator/Entry.php

```diff
@@ -0,0 +1,471 @@
+<?php
+
+/**
+ * This file is part of the contentful-management.php package.
+ *
+ * @copyright 2015-2017 Contentful GmbH
+ * @license   MIT
+ */
+
+namespace Contentful\Management\CodeGenerator;
+
+use Contentful\Link;
+use Contentful\Management\ApiDateTime;
+use Contentful\Management\Resource\Asset;
+use Contentful\Management\Resource\ContentType;
+use Contentful\Management\Resource\ContentType\Field\ArrayField;
+use Contentful\Management\Resource\ContentType\Field\FieldInterface;
+use Contentful\Management\Resource\ContentType\Field\LinkField;
+use Contentful\Management\Resource\ContentType\Validation\LinkContentTypeValidation;
+use Contentful\Management\Resource\Entry as EntryResource;
+use Contentful\Management\Resource\ResourceInterface;
+use PhpParser\Node;
+use PhpParser\Node\Stmt\Class_;
+use PhpParser\Node\Stmt\ClassMethod;
+
+/**
+ * Entry class.
+ */
+class Entry extends BaseCodeGenerator
+{
+    /**
+     * @var array
+     */
+    private $uses = [];
```

Comment:
Can you give a more detailed comment about what this property is?

Reply:
In PHP when you use a class you either use the FQCN, or add a `use` statement (usually at the beginning) and then employ only the actual class name.
This array is actually for keeping track which classes need to be imported (`use`d). It gets reset to a default state in `setDefaultUses`, and then throughout the code some methods might set one of the values in the array to true:

``` php
if ($usesAsset) {
    $this->uses['asset'] = true;
}
```

Eventually, if the `$this->uses['asset']` property will have the value `true`, it'll mean that the code has reached a path where it's generating a method that needs to have the `Contentful\Management\Resource\Asset` class imported.

Eventually, after the methods have been generated, this array will contain flags about all classes that need to be imported. Does this answer your question? 🙂